src/generic/generic-game-factory.vala

```diff
@@ -0,0 +1,54 @@
+// This file is part of GNOME Games. License: GPLv3
```

Comment:
The commit messages can and should go up 72 chars long (by the GNOME commit message convention https://wiki.gnome.org/Git/CommitMessages) so it should looks like this:
```
generic: Add GenericUriGameFactory

This class abstracts the process of creating a game from a uri, so
MimeTypeTrackerQuery does not have to deal with it.

This is part of a series of commits to add multi-source support.
```